starlette/datastructures.py

```diff
@@ -631,7 +631,7 @@ def setdefault(self, key: str, value: str) -> str:
         self._list.append((set_key, set_value))
         return value
 
-    def update(self, other: dict) -> None:
+    def update(self, other: typing.Mapping[str, str]) -> None:
```

Comment:
Is this required for this PR? I'd think long term we want do implement this the same as in httpx:
https://github.com/encode/httpx/blob/321d4aa5097fe7f24cdfed7191c44de589294780/httpx/_models.py#L965-L970

Reply:
this one was for consistency, but it's not part of  `Response` constructor, I can remove it.

I agree with you long term plan, but it's much more involved than this fix :)

for the funny story,  I noticed this issue by passing a  `httpx.Response.header` (which is a  `typing.MultableMapping`)  to `starlette.Response(header=...)`

Reply:
> this one was for consistency, but it's not part of Response constructor, I can remove it.

yeah since it's not directly related I would punt it until a future PR, but that's just my personal opinion, not a blocker for me

> for the funny story, I noticed this issue by passing a httpx.Response.header (which is a typing.MultableMapping) to starlette.Response(header=...)

hmm @tomchristie seems like something we should (eventually) fix

Reply:
removed :)

the PR now changes only `Response` & its subclasses